File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int n = s.size();
        int left = 0;
        int right = 0;
        int subarr = 0;
        vector<int>freq(3,0);

        while(right < n){
            freq[s[right]-'a']++;
            while(freq[0] > 0 && freq[1] > 0 && freq[2] > 0){
                subarr+=n-right;
                freq[s[left]-'a']--;
                left++;
            }

            right++;

        }
        return subarr;
        
    }
};

File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.py
class Solution(object):
    def numberOfSubstrings(self, s):
        right = 0
        left = 0
        sub = 0
        n = len(s)
        freq = [0,0,0];
        while(right < n):
            freq[ord(s[right])-ord('a')]+=1;
            while all(freq):
                sub+=n-right;
                freq[ord(s[left])-ord('a')]-=1;
                left+=1
            right+=1;
        return sub;


        """
        :type s: str
        :rtype: int
        """
        